app/serializers/v1/charge_filter_serializer.rb

```diff
@@ -5,9 +5,25 @@ class ChargeFilterSerializer < ModelSerializer
     def serialize
       {
         invoice_display_name: model.invoice_display_name,
-        properties: model.properties,
+        properties:,
         values: model.to_h
       }
     end
+
+    private
+
+    # TODO(pricing_group_keys): remove after deprecation of grouped_by
+    def properties
+      attributes = model.properties
+      if attributes["grouped_by"].present? && attributes["pricing_group_keys"].blank?
+        attributes["pricing_group_keys"] = attributes["grouped_by"]
+      end
+
+      if attributes["pricing_group_keys"].present? && attributes["grouped_by"].blank?
+        attributes["grouped_by"] = attributes["pricing_group_keys"]
+      end
```

Comment:
I'm trying to think of a case where you want `grouped_by` to hold a different value from `princing_group_keys`. Maybe we should for `grouped_by` value as soon as `princing_group_keys` is set

```suggestion
      if attributes["pricing_group_keys"].present?
        attributes["grouped_by"] = attributes["pricing_group_keys"]
      end
```

Similarily, we could add a check in the API so only one of the 2 values can be set and reject calls with both `grouped_by` and `pricing_group_keys`. It's probably not intentional to send both.